flink/src/test/java/org/apache/iceberg/flink/data/vectorized/TestVectorizedReads.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.data.vectorized;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.flink.FlinkCatalogTestBase;
+import org.apache.iceberg.flink.FlinkTableOptions;
+import org.apache.iceberg.flink.TestHelpers;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestVectorizedReads extends FlinkCatalogTestBase {
```

Comment:
Is possible to add a parameterized variable named `vectorized` in the previous  flink+ORC test cases , so that we could cover all the existing cases in the test suites ?   Let's follow the spark's test approach [in this PR](https://github.com/apache/iceberg/commit/6fab8f57bdb7e5fe7eadc3ff41558581338e1b69) if possible.

Reply:
We may need to add our own flink `MapColumnVector` in apache iceberg repo now so that we could pass all test cases after introducing the `vectorized` parameterized variable ( Because we've already had many test suites that introduced MAP data types, if the flink vectorized reader don't support MAP data type, then we have to pick those cases one by one,  It will be very trivial).

Reply:
I think we should add `MapColumnVector` in flink instead of iceberg,so I open  an issue in flink [FLINK-22591](https://issues.apache.org/jira/browse/FLINK-22591) , if  we add `MapColumnVector` in iceberg, and flink also add `MapColumnVector` in the futher, if the implementation is different, how do we resolve this conflict?